libraries/chain/webassembly/crypto.cpp

```diff
@@ -65,6 +66,50 @@ namespace eosio { namespace chain { namespace webassembly {
       }
    }
 
+   int32_t interface::recover_key_safe( legacy_ptr<const fc::sha256> digest,
+                                        legacy_span<const char> sig,
+                                        legacy_ptr<char> pub,
+                                        legacy_ptr<uint32_t> publen) const {
```

Comment:
These do not have to be `legacy_`. `const fc::sha256* digest, span<const char> sig, span<char> pub, uint32_t* publen`

Reply:
Failed to compile because `const fc::sha256*` is not whitelisted type.

Reply:
`legacy_*` should never be used for new intrinsics.  It has extremely weird behavior when there are multiple output parameters, due to possibility of aliasing.  Host functions with multiple output parameters using the legacy system require special manual control to ensure consistent behavior.  Also, a buffer must always use `span` so that the implementation knows to check memory bounds.  Failure to do so results in undefined behavior.